### PR TITLE
[otbn,doc] Add a diagram showing how the SV <-> Python bridge works

### PR DESCRIPTION
**Pull request by rswarbrick**
This is a bit complicated and I've explained it in person a couple of
times so it probably makes sense to document properly.

### CODE REVIEW

File: hw/ip/otbn/doc/dv/index.md [hunk 1]
```diff
@@ -52,7 +52,11 @@ The idle and interrupt signals are modelled with the basic
 
 As well as instantiating OTBN, the testbench also instantiates an `otbn_core_model`.
 This module wraps an ISS (instruction set simulator) subprocess and performs checks to make sure that OTBN behaves the same as the ISS.
-The model communicates with the testbench through an `otbn_model_if` interface, which is monitored by the `otbn_model_agent`, described below.
+The `otbn_core_model` module communicates with test sequences through an `otbn_model_if` interface, which is monitored by the `otbn_model_agent`, described below.
+The module communicates with the Python subprocess as shown in the diagram below
```
Comment (imphil) [inline hw/ip/otbn/doc/dv/index.md:56]:
```suggestion
The module communicates with the Python subprocess as shown in the diagram below.
```

**Review by imphil — APPROVED** [review]
In the figure: 
* "Sim.py" -> "sim.py" 
* I don't understand the "awkward C-ABI interface" (and it has two "interfaces" in it): Is it really an A*B*I? Maybe just remove this sentence.

**Comment by rswarbrick** [comment]
Huh! I mis-remembered and actually thought we'd capitalised Python files to match the classes in them. Fixed :-D

About the "awkward C-ABI interface" bit: The point is that the code is a bit awkward because we can't do things like call C++ methods directly (yes, I know why!). But, as you say, maybe that's a bit confusing. I'll drop the stuff in parens.

**Comment by sriyerg** [comment]
minor nit: Sim.py -> sim.py in the diagram.

File: hw/ip/otbn/doc/dv/model.svg [hunk 2]
```diff
@@ -0,0 +1,770 @@
...
+         sodipodi:role="line"
+         x="630"
+         y="170"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-family:'Liberation Sans';-inkscape-font-specification:'Liberation Sans'"
+         id="tspan2241">Together with the other Python files in the sim/</tspan><tspan
+         sodipodi:role="line"
+         x="630"
+         y="191.60423"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-family:'Liberation Sans';-inkscape-font-specification:'Liberation Sans'"
+         id="tspan2265">directory, Sim.py forms the core of the instruction</tspan><tspan
```
Comment (imphil) [inline hw/ip/otbn/doc/dv/model.svg:614]: here :-)

Comment (rswarbrick) [reply hw/ip/otbn/doc/dv/model.svg:614]: Oh damn. I fixed one of them...

**Comment by rswarbrick** [comment]
Fixed properly in https://github.com/lowRISC/opentitan/pull/8265.

**Pull request merged**